spec/models/foreign_investor_spec.rb

```diff
@@ -0,0 +1,33 @@
+require './spec/spec_helper'
+
+describe ForeignInvestor do
+  let(:player) { Player.new 1, 'Test' }
+  let(:company) { Company.new player, 'BME', 'Bergisch', :red, 1, 1, [] }
+  let(:company2) { Company.new player, 'AKE', 'Altona-Kieler Eisenbahn-Gesellschaft', :red, 6, 2, [] }
+  let(:share_price) { SharePrice.initial_market[6] } # 10, 6
+  let(:corporation) { Corporation.new 'Android', company, share_price, SharePrice.initial_market }
+  subject { ForeignInvestor.new }
+
+  it 'should init' do
+    expect(subject).not_to be_nil
+  end
+
+  describe '#initialize' do
+    it 'should set variables correctly' do
+      expect(subject.companies).not_to be_nil
+      expect(subject.cash).to eq(0)
+    end
+  end
+
+  describe '#purchase_companies' do
+    it "should buy only one company if it has 5 dollars" do
+      # need to draw companies in game.
+      @companies = []
+      @companies[0] = company
```

Comment:
This was a statement about the test as it was set up.  I gave him only 5 dollars, and there were two companies available.   A 1 dollar company and a 6 dollar company.  In this case, it would only be able to purchase a single company.